cartosky/core.py

```diff
@@ -124,7 +125,20 @@ def draw_hpxmap(self, hpxmap, pixel=None, nside=None, xsize=800,
         im,lon,lat,values : mpl image with pixel longitude, latitude (deg), and values
         """
 
-        healpix.check_hpxmap(hpxmap,pixel,nside)
+        is_hsp = healpix.check_hpxmap(hpxmap,pixel,nside)
+        if is_hsp:
+            if nside is not None and nside < hpxmap.nside_sparse:
+                hpxmap.degrade(nside)
+            elif nside is not None and nside > hpxmap.nside_sparse:
+                return NotImplementedError("Map upgrading is not implemented, try with different nside")
+            else:
+                nside = hpxmap.nside_sparse
+            if pixel is not None:
+                hpxmap = hpxmap.get_values_pix(hp.ring2nest(nside, pixel))
+            else:
+                _hpxmap = hpxmap.get_values_pix(hpxmap.valid_pixels)
+                pixel = hp.nest2ring(nside, hpxmap.valid_pixels)
+                hpxmap = _hpxmap
```

Comment:
Do we really want to implicitly degrade the map for the user? Also, won't this degrade the object that was passed in inplace? 

The current implementation takes *either* `hpxmap` *or* requires all three of `hpxmap, pixel, nside`. I think that we'd like to maintain this.

Reply:
I haven't looked at the code in detail, but this did strike me.  We want to avoid `degrade` because that's a slow and potentially memory-expensive operation.

Reply:
Yes, that degrades the object inplace. Yes, that is only bypassed if the map is a healsparse map but I can keep the same functionality.